clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClientTest.java

```diff
@@ -268,26 +270,36 @@ public void testMetadataFailurePropagated() {
     public void testFutureCompletionOutsidePoll() throws Exception {
         // Tests the scenario in which the request that is being awaited in one thread
         // is received and completed in another thread.
+        
+        final CountDownLatch t1TheardCountDownLatch = new CountDownLatch(1);
+        final CountDownLatch t2ThreadCountDownLatch = new CountDownLatch(2);
 
         final RequestFuture<ClientResponse> future = consumerClient.send(node, heartbeat());
         consumerClient.pollNoWakeup(); // dequeue and send the request
 
         client.enableBlockingUntilWakeup(2);
-        Thread t1 = new Thread(() -> consumerClient.pollNoWakeup());
+        Thread t1 = new Thread(() -> assertDoesNotThrow(() -> {
+            t1TheardCountDownLatch.countDown();
+            consumerClient.pollNoWakeup();
+            t2ThreadCountDownLatch.countDown();
+        }));
+        
         t1.start();
 
-        // Sleep a little so that t1 is blocking in poll
-        Thread.sleep(50);
-
-        Thread t2 = new Thread(() -> consumerClient.poll(future));
+        Thread t2 = new Thread(() -> assertDoesNotThrow(() -> {
```

Comment:
ditto

Reply:
`t2ThreadCountDownLatch.await()` need to catch the `InterruptedException`, I think use `assertDoesNotThrow` will be more clearly

Reply:
However, it is a no-op since the error happens in a separate thread, right?